plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts

```diff
@@ -203,6 +203,13 @@ export class MySQLInstrumentation extends InstrumentationBase<
 
   private _getConnectionCallbackPatchFn(cb: Function, format: formatType) {
     const thisPlugin = this;
+    // If a parent span exists, bind the callback so that the parent span
+    // is still visible after we've gone through the MySQL pool code.
+    const activeContext = context.active();
+    const parentSpan = trace.getSpan(activeContext);
+    if (parentSpan && typeof cb === 'function') {
```

Comment:
I don't believe you should check for span here, context can propagate other things (like baggage) without necessarily having a span, checking that cb is a fn should be enough !

Reply:
Also you could move the bind to when the callback is actually called line 226

Reply:
@vmarchaud thanks for taking a look! I've removed the check for the parent span, and used `with` instead of `bind` at the point that the callback gets called - I think `with` makes more sense now?

Reply:
@sstone1 Yes, lgtm now ! thanks for taking the time to fix this :)